packages/next/telemetry/events/build.ts

```diff
@@ -19,13 +19,24 @@ type EventBuildOptimized = {
   totalPageCount: number
   staticPageCount: number
   ssrPageCount: number
+  hasInvalidPages: boolean
 }
 
+const INVALID_PAGES = /^[\\/]__(tests|generated)__[\\/]/
+const TEST_FILE = /\.test\.[jt]sx?$/
+
 export function eventBuildOptimize(
-  event: EventBuildOptimized
+  pagePaths: string[],
+  event: Omit<EventBuildOptimized, 'totalPageCount' | 'hasInvalidPages'>
 ): { eventName: string; payload: EventBuildOptimized } {
   return {
     eventName: EVENT_BUILD_OPTIMIZE,
-    payload: event,
+    payload: {
+      ...event,
+      totalPageCount: pagePaths.length,
+      hasInvalidPages: pagePaths.some(
```

Comment:
I think this is a bad event name—when I hear invalid page, I think not exporting a React Component.

Can we name this something explicit like `hasDunderPages` and `hasTestPages`?

Reply:
`hasTestPages` sounds good to me, it also includes `__generated__` but anyway, I can't think of a better name for this.